File: src/entities/Business.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, CreateDateColumn } from 'typeorm';
import { User } from './User';
import { BusinessModule } from './BusinessModule';
import { StoreOrder } from './StoreOrder';
import { MediaFile } from './MediaFile';
import { ActivityLog } from './ActivityLog';
import { IncomeExpense } from './IncomeExpense';

@Entity()
export class Business {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ unique: true })
  name: string;

  @Column()
  sector: string;

  @OneToMany(() => User, (user) => user.business)
  users: User[];

  @OneToMany(() => BusinessModule, (bm) => bm.business)
  businessModules: BusinessModule[];

  @OneToMany(() => StoreOrder, (order) => order.business)
  storeOrders: StoreOrder[];

  @OneToMany(() => MediaFile, (media) => media.business)
  mediaFiles: MediaFile[];

  @OneToMany(() => ActivityLog, (log) => log.business)
  activityLogs: ActivityLog[];

  @OneToMany(() => IncomeExpense, (ie) => ie.business)
  incomeExpenses: IncomeExpense[];

  @CreateDateColumn()
  createdAt: Date;
} 

File: Entities/Business.cs
using System;
using System.Collections.Generic;

namespace ModularSaaS.Entities
{
    public class Business
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Sector { get; set; }
        public Guid? OwnerId { get; set; }
        public DateTime? SubscriptionStart { get; set; }
        public DateTime? SubscriptionEnd { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<User> Users { get; set; }
        public ICollection<BusinessModule> BusinessModules { get; set; }
        public ICollection<StoreOrder> StoreOrders { get; set; }
        public ICollection<MediaFile> MediaFiles { get; set; }
        public ICollection<ActivityLog> ActivityLogs { get; set; }
        public ICollection<IncomeExpense> IncomeExpenses { get; set; }
    }
} 